dynamodb/src/main/scala/monix/connect/dynamodb/DynamoDbSubscriber.scala

```diff
@@ -20,42 +20,65 @@ package monix.connect.dynamodb
 import monix.eval.Task
 import monix.execution.cancelables.AssignableCancelable
 import monix.execution.{Ack, Callback, Scheduler}
-import monix.reactive.observers.Subscriber
 import monix.reactive.Consumer
 import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
 import software.amazon.awssdk.services.dynamodb.model.{DynamoDbRequest, DynamoDbResponse}
+import monix.reactive.observers.Subscriber
 
 import scala.concurrent.Future
 
-private[dynamodb] class DynamoDbSubscriber[In <: DynamoDbRequest, Out <: DynamoDbResponse]()(
+private[dynamodb] class DynamoDbSubscriber[In <: DynamoDbRequest, Out <: DynamoDbResponse](retriesPerOp: Int)(
   implicit
   dynamoDbOp: DynamoDbOp[In, Out],
   client: DynamoDbAsyncClient)
   extends Consumer[In, Out] {
 
+  require(retriesPerOp >= 0, "Retries per operation must be higher or equal than 0.")
+
   override def createSubscriber(cb: Callback[Throwable, Out], s: Scheduler): (Subscriber[In], AssignableCancelable) = {
     val sub = new Subscriber[In] {
 
       implicit val scheduler = s
-      private var dynamoDbResponse: Task[Out] = _
-
-      def onNext(dynamoDbRequest: In): Future[Ack] = {
-        dynamoDbResponse = Task.from(dynamoDbOp.execute(dynamoDbRequest))
+      private var dynamoDbResponse: Out = _
 
-        dynamoDbResponse.onErrorRecover { case _ => monix.execution.Ack.Stop }
-          .map(_ => monix.execution.Ack.Continue)
-          .runToFuture
+      def onNext(request: In): Future[Ack] = {
+        synchronized {
```

Comment:
I think `FoldLeftTaskConsumer` is synchronized because variables can be modified from `Cancelable` (which is called by who knows who).

`dynamoDbResponse` is only used in `onNext` / `onComplete` so I don't think it's needed, the protocol guarantees that they will always see the latest `dynamoDbResponse`. 

`consumeWith` also wraps subscriber in `SafeSubscriber` which ensures that `onComplete` won't be called until the last `onNext` (last request in our case) has completed.